### ISSUE TITLE
macos/Flutter/GeneratedPluginRegistrant.swift should be created in macos/Flutter/ephemeral/GeneratedPluginRegistrant.swift

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#71762]
Generated files should be created in the `Flutter/ephemeral` directory.

**Comment by stuartmorgan** [comment]
No, not unless something has changed in the plugin code. The way all existing platforms worked when I set this up for desktop was that the plugin code was generated, but expected to be checked in. I don't know why, but that's how they were set up, and I'm not at all confident that there aren't ways to get into a state where the generation of that file is optimized away, and build will therefore fail.

I think the fact that we require checking in generated code is poor design, but really didn't want to diverge desktop on something that subtle.

**Comment by stuartmorgan** [comment]
> Generated files should be created in the Flutter/ephemeral directory.

The precise statement of the intent of that folder on desktop is "files which we do not want checked in should be created in the Flutter/ephemeral directory". (Which per the above would ideally be the same statement, but isn't.)

**Comment by jmagman** [comment]
Hm. it's not checked into `flutter_gallery` and that has plugins, and it builds.
https://github.com/flutter/flutter/tree/master/dev/integration_tests/flutter_gallery/macos/Flutter

**Comment by stuartmorgan** [comment]
Clean builds will definitely work. Others... 🤷 

*Maybe* there's currently no case where we skip that generation (I'd have to go read it and I'm on a phone). But there is code in the plugins path that decides whether plugins have changed and skips some steps of they haven't, and if that includes the generated plugin code on all platforms--or ever does in the future--there would be breakage only in certain kinds of builds.

If we want to change all platforms to not need to check in generated plugin code, and cover that thoroughly with tests, I'm all for it. But I'm *strongly* against changing it for only macOS, or only desktop in general.
